### PR TITLE
Release 2.4: port high MMIO support

### PR DESCRIPTION
**Pull request by lifeix**
port high MMIO support from master to release_2.4

Signed-off-by: Li Fei1 <fei1.li@intel.com>
Tracked-On: #5788

### CODE REVIEW

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br>No New Function Declaration/Definition Mismatch

**Comment by gvancuts** [comment]
The `make doc` should pass if we backport existing commits into the `release_2.4`. I believe the following ones are needed:
* 31a9d053f doc: update our FAQ section
* 0bd384d41 doc: fix all headings to use title case
* c741468b9 doc: remove Kconfig reference documentation

File: hypervisor/arch/x86/page.c [hunk 1]
```diff
@@ -13,24 +13,82 @@
...
+#define PPT_PDPT_PAGE_NUM	PDPT_PAGE_NUM(MAX_PHY_ADDRESS_SPACE)
+#define PPT_PD_PAGE_NUM	PD_PAGE_NUM(MAX_PHY_ADDRESS_SPACE)
+#define PPT_PT_PAGE_NUM	0UL	/* not support 4K granularity page mapping */
+/* must be a multiple of 64 */
+#define PPT_PAGE_NUM	(roundup((PPT_PML4_PAGE_NUM + PPT_PDPT_PAGE_NUM + \
+			PPT_PD_PAGE_NUM + PPT_PT_PAGE_NUM), 64U))
+static struct page ppt_pages[PPT_PAGE_NUM];
+static uint64_t ppt_page_bitmap[PPT_PAGE_NUM / 64];
+
+/* ppt: pripary page pool */
```
[the remark this answers is not in the document]
Comment (lifeix) [reply hypervisor/arch/x86/page.c:32]: Good catch. Will fix this typo. Thanks, gvancuts.

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br>No New Function Declaration/Definition Mismatch

**Comment by wenlingz** [comment]
@fyin1 please help approve this PR, thanks

**Review by fyin1 — APPROVED** [review]

**Pull request merged**